odl/solvers/functional/default_functionals.py

```diff
@@ -2193,6 +2198,169 @@ def __repr__(self):
                                          self.__norm.pwisenorm.exponent)
 
 
+class IndicatorSimplex(Functional):
+
+    r"""The indicator functional of a simplex in any dimension.
+
+    Notes
+    -----
+    The functional :math:`F` is given by:
+
+    .. math::
+        F(x)
+        =
+        \begin{cases}
+            0 & \text{if } x_i \geq 0 \forall i \text{ and } \sum_i x_i = r \\
```

Comment:
Isn't this a non-convex functional? Wouldn't it be better to only require that the sum is less than or equal to r?

Reply:
This is convex as the simplex is a convex set. For deconvolution it is very important to have the sum equal to r in order to get non-trivial solutions.